omod/src/main/java/org/openmrs/module/spa/filter/SpaFilter.java

```diff
@@ -22,21 +22,28 @@
 
 import org.apache.commons.lang3.StringUtils;
 import org.openmrs.api.context.Context;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import static org.openmrs.module.spa.SpaConstants.DEFAULT_SPA_BASE_URL;
 import static org.openmrs.module.spa.SpaConstants.GP_KEY_SPA_BASE_URL;
 
 public class SpaFilter implements Filter {
 
+	private static final Logger logger = LoggerFactory.getLogger(SpaFilter.class);
+
 	@Override
 	public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain) throws IOException, ServletException {
 		
 		HttpServletRequest request = (HttpServletRequest) req;
-		String spaBasePath = Context.getAdministrationService().getGlobalProperty(GP_KEY_SPA_BASE_URL,
-				DEFAULT_SPA_BASE_URL);
-		
-		if (StringUtils.isBlank(spaBasePath)) {
-			spaBasePath = DEFAULT_SPA_BASE_URL;
+
+		String spaBasePath = DEFAULT_SPA_BASE_URL;
+
+		try {
+			spaBasePath = Context.getAdministrationService().getGlobalProperty(GP_KEY_SPA_BASE_URL,
+					DEFAULT_SPA_BASE_URL);
+		} catch (Exception e) {
```

Comment:
Can we catch the specific exception instead of the generic one?

Reply:
Thinking about this again, will some one be able to use the spa module if they override the default base url?